paderbox/io/play.py

```diff
@@ -20,29 +21,35 @@ def play(
         name=None,
         stereo=False,
         normalize=True,
+        display=True,
 ):
-    """ Tries to guess, what the input data is. Plays time series and stft.
+    """
+    Tries to guess, what the input data is. Plays time series and stft.
 
     Provides an easy to use interface to play back sound in an IPython Notebook.
 
-    :param data: Time series with shape (frames,)
-        or stft with shape (frames, channels, bins) or (frames, bins)
-        or string containing path to audio file.
-    :param channel: Channel, if you have a multichannel stft signal or a
-        multichannel audio file.
-    :param sample_rate: Sampling rate in Hz.
-    :param size: STFT window size
-    :param shift: STFT shift
-    :param window: STFT analysis window
-    :param scale: Scale the Volume, currently only amplification with clip
-        is supported.
-    :param name: If name is set, then in ipynb table with name and audio is
-                 displayed
-    :param stereo: If set to true, you can listen to channel as defined by
-        `channel` parameter and the next channel at the same time.
-    :param normalize: It true, normalize the data to have values in the range
-        from 0 to 1. Can only be disabled with newer IPython versions.
-    :return:
+    Args:
+        data: Time series with shape (frames,)
+            or stft with shape (frames, channels, bins) or (frames, bins)
+            or string containing path to audio file.
+        channel: Channel, if you have a multichannel stft signal or a
+            multichannel audio file.
+        sample_rate: Sampling rate in Hz.
+        size: STFT window size
+        shift: STFT shift
+        window: STFT analysis window
+        window_length: STFT window_length
+        scale: Scale the Volume, currently only amplification with clip
+            is supported.
+        name: If name is set, then in ipynb table with name and audio is
+            displayed
+        stereo: If set to true, you can listen to channel as defined by
+            `channel` parameter and the next channel at the same time.
+        normalize: It true, normalize the data to have values in the range
+            from 0 to 1. Can only be disabled with newer IPython versions.
+        display: When True, display the audio, otherwise return the widget.
+    Returns:
+
     """
     if isinstance(data, dict):
```

Comment:
Just as a suggestion: Support nested dicts and lists, e.g., with `FlatView`? That way one could pass an entire `example['audio_data']` dict to the function and get a nice view in the notebook

Reply:
While I like the idea to have a nice view, it won't create one for `example['audio_data']`, it would simply pollute the notebook (i.e. one play for each channel for each audio file, which may break for long source files).
I think in most cases it is better to call it manually, this function has already more magic as I like.